### PR TITLE
Add E2E acknowledgements

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Adds Data Prepper documentation for E2E acknowledgements.

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/configuration/sources/s3.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ buffer_timeout | No | Duration | The amount of time allowed for for writing even
 records_to_accumulate | No | Integer | The number of messages that accumulate before writing to the buffer. Default value is 100.
 metadata_root_key | No | String | Base key for adding S3 metadata to each Event. The metadata includes the key and bucket for each S3 object. Defaults to `s3/`.
 disable_bucket_ownership_validation | No | Boolean | If `true`, the S3Source will not attempt to validate that the bucket is owned by the expected account. The expected account is the same account that owns the Amazon SQS queue. Defaults to `false`.
+acknowledgements | No | Boolean | If `true`, enables S3 sources to receive [end-to-end acknowledgements]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines/#end-to-end-acknowledgements) when events are received by OpenSearch sinks.
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _data-prepper/pipelines/configuration/sources/s3.md]: Done

**Review by ariamarble — APPROVED** [review]

File: _data-prepper/pipelines/configuration/sources/s3.md [hunk 2]
```diff
@@ -87,4 +88,4 @@ The `s3` processor includes the following metrics.
 
 * `s3ObjectSizeBytes`: Measures the size of Amazon S3 objects as reported by the Amazon S3 `Content-Length`. For compressed objects, this is the compressed size.
 * `s3ObjectProcessedBytes`: Measures the bytes processed by the `s3` source for a given object. For compressed objects, this is the uncompressed size.
-* `s3ObjectsEvents`: Measures the number of events (sometimes called records) produced by an Amazon S3 object.
\ No newline at end of file
+* `s3ObjectsEvents`: Measures the number of events (sometimes called records) produced by an Amazon S3 object.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/s3.md:89]:
```suggestion
* `s3ObjectsEvents`: Measures the number of events (sometimes called records) produced by an S3 object.
```

File: _data-prepper/pipelines/pipelines.md [hunk 3]
```diff
@@ -48,6 +48,16 @@ simple-sample-pipeline:
 Starting from Data Prepper 2.0, you can define pipelines across multiple configuration YAML files, where each file contains the configuration for one or more pipelines. This gives you more freedom to organize and chain complex pipeline configurations. For Data Prepper to load your pipeline configuration properly, place your configuration YAML files in the `pipelines` folder under your application's home directory (e.g. `/usr/share/data-prepper`).
 {: .note }
 
+## End-to-end acknowledgments
+
+Data Prepper ensures the durability and reliability of data written from sources and delivered to sinks through end-to-end (E2E) acknowledgments. The E2E acknowledgment begins at the source, which monitors a batch of events set inside pipelines and waits for a positive acknowledgment when those events are successfully pushed to sinks. When a pipeline contains multiple sinks, including sinks set as additional Data Prepper pipelines, the E2E acknowledgment sends when events are received by the final sink in a pipeline chain.
```
Comment (natebower) [inline _data-prepper/pipelines/pipelines.md:53]:
```suggestion
Data Prepper ensures the durability and reliability of data written from sources and delivered to sinks through end-to-end (E2E) acknowledgments. An E2E acknowledgment begins at the source, which monitors a batch of events set inside pipelines and waits for a positive acknowledgment when those events are successfully pushed to sinks. When a pipeline contains multiple sinks, including sinks set as additional Data Prepper pipelines, the E2E acknowledgment sends when events are received by the final sink in a pipeline chain.
```

File: _data-prepper/pipelines/pipelines.md [hunk 4]
```diff
@@ -48,6 +48,16 @@ simple-sample-pipeline:
...
+
+Alternatively, the source sends a negative acknowledgment when an event cannot be delivered to a sink for any reason. 
+
+When any component of a pipeline fails and is unable to send an event, the source receives no acknowledgment. In the case of A failure, the pipeline's source times out. This gives you the ability to take any necessary actions to address the source failure, including rerunning the pipeline or logging the failure.
```
Comment (natebower) [inline _data-prepper/pipelines/pipelines.md:57]:
```suggestion
When any component of a pipeline fails and is unable to send an event, the source receives no acknowledgment. In the case of a failure, the pipeline's source times out. This gives you the ability to take any necessary actions to address the source failure, including rerunning the pipeline or logging the failure.
```

File: _data-prepper/pipelines/configuration/sources/s3.md [hunk 5]
```diff
@@ -1,16 +1,14 @@
...
 nav_order: 20
 ---
 
-# s3
+# `s3` source
 
-## Overview
-
-`s3` is a source plugin that reads events from [Amazon Simple Storage Service (S3)](https://aws.amazon.com/s3/) (Amazon S3) objects. The following table describes options you can use to configure the `s3` source.
+The Amazon Simple Storage Service (S3) source plugin reads events from [S3](https://aws.amazon.com/s3/) objects. The following table describes options you can use to configure the `s3` source.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/s3.md:11]:
```suggestion
The Amazon Simple Storage Service (Amazon S3) source plugin reads events from [S3](https://aws.amazon.com/s3/) objects. The following table describes options you can use to configure the `s3` source.
```

File: _data-prepper/pipelines/configuration/sources/s3.md [hunk 6]
```diff
@@ -24,6 +22,7 @@ buffer_timeout | No | Duration | The amount of time allowed for for writing even
 records_to_accumulate | No | Integer | The number of messages that accumulate before writing to the buffer. Default value is 100.
 metadata_root_key | No | String | Base key for adding S3 metadata to each Event. The metadata includes the key and bucket for each S3 object. Defaults to `s3/`.
 disable_bucket_ownership_validation | No | Boolean | If `true`, the S3Source will not attempt to validate that the bucket is owned by the expected account. The expected account is the same account that owns the Amazon SQS queue. Defaults to `false`.
+acknowledgments | No | Boolean | If `true`, enables S3 sources to receive [end-to-end acknowledgments]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines/#end-to-end-acknowledgments) when events are received by OpenSearch sinks.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/s3.md:25]: `s3` sources?

File: _data-prepper/pipelines/pipelines.md [hunk 7]
```diff
@@ -48,6 +48,16 @@ simple-sample-pipeline:
...
+
+As of Data Prepper 2.2, only the S3 source and OpenSearch Sink support E2E acknowledgments.
```
Comment (natebower) [inline _data-prepper/pipelines/pipelines.md:59]: `s3` source?

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

File: _data-prepper/pipelines/configuration/sources/s3.md (hunk 6 above)
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/sources/s3.md:25]:
```suggestion
acknowledgments | No | Boolean | If `true`, enables `s3` sources to receive [end-to-end acknowledgments]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines/#end-to-end-acknowledgments) when events are received by OpenSearch sinks.
```

File: _data-prepper/pipelines/pipelines.md [hunk 8]
```diff
@@ -48,6 +48,16 @@ simple-sample-pipeline:
...
 
+## End-to-end acknowledgments
+
+Data Prepper ensures the durability and reliability of data written from sources and delivered to sinks through end-to-end (E2E) acknowledgments. An E2E acknowledgment begins at the source, which monitors a batch of events set inside pipelines and waits for a positive acknowledgment when those events are successfully pushed to sinks. When a pipeline contains multiple sinks, including sinks set as additional Data Prepper pipelines, the E2E acknowledgment sends when events are received by the final sink in a pipeline chain.
+
+Alternatively, the source sends a negative acknowledgment when an event cannot be delivered to a sink for any reason. 
+
+When any component of a pipeline fails and is unable to send an event, the source receives no acknowledgment. In the case of a failure, the pipeline's source times out. This gives you the ability to take any necessary actions to address the source failure, including rerunning the pipeline or logging the failure.
+
+As of Data Prepper 2.2, only the S3 source and OpenSearch Sink support E2E acknowledgments.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/pipelines.md:59]:
```suggestion
As of Data Prepper 2.2, only the `s3` source and `opensearch` sink support E2E acknowledgments. 
```

**Review by JeffH-AWS — APPROVED** [review]

**Pull request merged**